apps/maestro/src/server/routers/erc20/getERC20TokenBalanceForOwner.ts

```diff
@@ -16,11 +17,68 @@ export const getERC20TokenBalanceForOwner = publicProcedure
   .input(
     z.object({
       chainId: z.number(),
-      tokenAddress: hex40Literal(),
-      owner: hex40Literal(),
+      tokenAddress: z.string(),
+      owner: z.string(),
     })
   )
   .query(async ({ input, ctx }) => {
+    // Sui address length is 66
+    if (input.tokenAddress?.length === 66) {
+      let isTokenOwner = false;
+
+      const client = new SuiClient({ url: getFullnodeUrl("testnet") }); // TODO: make this configurable
+
+      // Get the coin type
+      const modules = await client.getNormalizedMoveModulesByPackage({
+        package: input.tokenAddress,
+      });
+      const coinSymbol = Object.keys(modules)[0];
+      const coinType = `${input.tokenAddress}::${coinSymbol?.toLocaleLowerCase()}::${coinSymbol?.toUpperCase()}`;
```

Comment:
```suggestion
      const coinType = `${input.tokenAddress}::${coinSymbol?.toLowerCase()}::${coinSymbol?.toUpperCase()}`;
```
don't think we need locale here.